### PR TITLE
Use Vars.substituend in CClosure.comp_subs

### PR DESCRIPTION
**Pull request by SkySkimmer**

### CODE REVIEW

**Comment by SkySkimmer** [comment]
@coqbot bench

**Comment by SkySkimmer** [comment]
@coqbot run full ci

**Comment by SkySkimmer** [comment]
It's a small improvement but also a small code change

File: kernel/cClosure.ml [hunk 1]
```diff
@@ -602,7 +602,7 @@ and to_constr_case lfts ci u pms (p,r) iv c ve env =
             Array.map f_ctx ve)
 
 and comp_subs el (s,u') =
-  Esubst.lift_subst (fun el c -> lazy (to_constr el c)) el s, u'
+  Esubst.lift_subst (fun el c -> lazy (Vars.make_substituend @@ to_constr el c)) el s, u'
```
Comment (ppedrot) [inline kernel/cClosure.ml:605]: We could probably be even more clever here by making to_constr return a substituend so as to share more lifts.

**Review by ppedrot — APPROVED** [review]

**Comment by ppedrot** [comment]
@coqbot merge now

**Pull request merged**